pkgs/games/0ad/game.nix

```diff
@@ -1,65 +1,65 @@
-{ stdenv, lib, perl, fetchurl, python2
-, pkg-config, spidermonkey_38, boost, icu, libxml2, libpng, libsodium
+{ stdenv, lib, perl, fetchurl, python2, fmt, libidn
+, pkg-config, spidermonkey_78, boost, icu, libxml2, libpng, libsodium
 , libjpeg, zlib, curl, libogg, libvorbis, enet, miniupnpc
 , openal, libGLU, libGL, xorgproto, libX11, libXcursor, nspr, SDL2
-, gloox, nvidia-texture-tools
-, withEditor ? true, wxGTK ? null
+, gloox, nvidia-texture-tools, zeroad-data
+, withEditor ? true, wxGTK
 }:
 
-assert withEditor -> wxGTK != null;
-
+# You can find more instructions on how to build 0ad here:
+#    https://trac.wildfiregames.com/wiki/BuildInstructions
+
+let
+  # the game requires a special version 78.6.0 of spidermonkey, otherwise
+  # we get compilation errors. We override the src attribute of spidermonkey_78
+  # in order to reuse that declartion, while giving it a different source input.
```

Comment:
I wonder if this is the correct solution. The second patch in the derivation for 0.0.23b removed the minor version check, so maybe that's the correct solution here as well.

Reply:
No, i tried to remove that check already and it gives a compilation error.

Reply:
The [Fedora patch](https://src.fedoraproject.org/rpms/0ad/raw/26dc1657f6e3c0ad9f1180ca38cd79b933ef0c8b/f/0ad-mozjs-incompatible.patch) contains two changes – if you remove the second change, which seems no longer necessary, and perhaps fix the line numbers to `@@ -81,19 +81,6 @@` it does apply. But I tend to agree that your solution of overriding the `spidermonkey_78` version is better, since 78.6 is the version that is tested.